tss-esapi/src/context/tpm_commands/signing_and_signature_verification.rs

```diff
@@ -45,9 +45,10 @@ impl Context {
         key_handle: KeyHandle,
         digest: Digest,
         scheme: SignatureScheme,
-        validation: HashcheckTicket,
+        validation: impl Into<Option<HashcheckTicket>>,
```

Comment:
I guess since the docs on this method are fairly lightweight, there's no point in specifying what happens if one passes `None` here?

Reply:
It's probably better to say something here since we're looking. Maybe something like:

"For signatures using a restricted key, a hashcheck must be provided. For unrestricted keys, this may be None"

Reply:
Yes, I should probably also update the documentation of the sign context method while I am at because it was seriously lacking.

Reply:
Please have a look at the documentation that I added and see if it seems alright.